build-tools/automation/azure-pipelines-oss.yaml

```diff
@@ -3,65 +3,188 @@
-
-  - script: make jenkins V=1 PREPARE_CI_PR=1 PREPARE_AUTOPROVISION=1 CONFIGURATION=$(OSSBuildConfiguration)
-    displayName: make jenkins
-
-  - script: make package-deb V=1 CONFIGURATION=$(OSSBuildConfiguration)
-    displayName: make package-deb
-
-  - script: >
-      mkdir -p $(System.DefaultWorkingDirectory)/bin/Build$(OSSBuildConfiguration)/linux-artifacts &&
-      cp $(System.DefaultWorkingDirectory)/*xamarin.android*.tar.bz2 $(System.DefaultWorkingDirectory)/bin/Build$(OSSBuildConfiguration)/linux-artifacts &&
-      cp $(System.DefaultWorkingDirectory)/*.changes $(System.DefaultWorkingDirectory)/bin/Build$(OSSBuildConfiguration)/linux-artifacts &&
-      cp $(System.DefaultWorkingDirectory)/*.dsc $(System.DefaultWorkingDirectory)/bin/Build$(OSSBuildConfiguration)/linux-artifacts &&
-      cp $(System.DefaultWorkingDirectory)/*.deb $(System.DefaultWorkingDirectory)/bin/Build$(OSSBuildConfiguration)/linux-artifacts
-    displayName: copy linux artifacts
-
-  - task: PublishPipelineArtifact@1
-    displayName: publish linux artifacts
-    inputs:
-      artifactName: Linux Packages
-      targetPath: $(System.DefaultWorkingDirectory)/bin/Build$(OSSBuildConfiguration)/linux-artifacts
-
-  - template: yaml-templates/upload-results.yaml
-    parameters:
-      configuration: $(OSSBuildConfiguration)
-      artifactName: OSS Build Results - Linux
+  EnableMacStage: true
+  EnableLinuxStage: false
+
+  XA.Build.MacOSSPool: VSEng-Xamarin-RedmondMacMojaveBuildPool-Android-OSS
+  XA.Build.LinuxOSSPool: Xamarin-Android-Ubuntu-Public
+
+  XA.Build.Configuration: Release
+
+  NuGetArtifactName: nupkgs
+  EXTRA_MSBUILD_ARGS: /p:AutoProvision=True /p:AutoProvisionUsesSudo=True /p:IgnoreMaxMonoVersion=False
+  PREPARE_FLAGS: PREPARE_CI=1 PREPARE_CI_PR=1
+  DotNetCoreVersion: 3.1.201
+  # Version number from: https://dotnet.microsoft.com/download/dotnet-core/5.0
+  DotNetCorePreviewVersion: 5.0.100-preview.4.20227.14
+
+  BUILD_TESTS: true
+  RUN_TESTS: false
+
+stages:
+- stage: mac_stage
+  displayName: Mac
+  condition: eq(variables['EnableMacStage'], 'true')  # The variable is defined on the pipeline definition
+  jobs:
+  - job: mac_build
+    displayName: Mac Build
+    pool:
+      name: $(XA.Build.MacOSSPool)  # The variable is defined on the pipeline definition
+    timeoutInMinutes: 180
+    cancelTimeoutInMinutes: 5
+    workspace:
+      clean: all
+
+    steps:
+    - checkout: self          # https://docs.microsoft.com/en-us/azure/devops/pipelines/yaml-schema?view=azure-devops&tabs=schema#checkout
+      path: xamarin-android
+      clean: true             # Executes: git clean -ffdx && git reset --hard HEAD
+      submodules: recursive
+
+    - template: yaml-templates/use-dot-net.yaml
+      parameters:
+        version: $(DotNetCorePreviewVersion)
+
+    - template: yaml-templates/use-dot-net.yaml
+      parameters:
+        version: $(DotNetCoreVersion)
+
+    # Prepare and build everything
+    - script: |
```

Comment:
Given d3e0fc831b98b5481f220fcd9c0967d096bf6ee8, I'm wary of using `script: |`, and changing this to use `script: >` should be straightforward.

Reply:
Changed to `script: >` per 4244cb01c8ff5b9ec43149f4da76ec982427e877